src/robotFactory.js

```diff
@@ -1,10 +1,77 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    const { x = 0, y = 0 } = coords;
 
-class FlyingRobot {}
+    this.name = name;
+    this.weight = weight;
 
-class DeliveryDrone {}
+    this.coords = {
+      x, y,
+    };
+    this.chipVersion = chipVersion;
+  }
+
+  goForward(step = 1) {
+    this.coords.y += step;
+  }
+
+  goBack(step = 1) {
+    this.coords.y -= step;
+  }
+
+  goRight(step = 1) {
+    this.coords.x += step;
+  }
+
+  goLeft(step = 1) {
+    this.coords.x -= step;
+  }
+
+  getInfo() {
+    return `Robot: ${this.name}, Chip version: ${this.chipVersion}, `
+    + `Weight: ${this.weight}`;
+  }
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    const { x = 0, y = 0, z = 0 } = coords;
+
+    super(name, weight, {
+      x, y,
+    }, chipVersion);
+    this.coords.z = z;
```

Comment:
No need to spred cords and create new object for basic robot constructor
```suggestion
    super(name, weight, coords, chipVersion);
    this.coords.z = coords.z || 0;
```